pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -907,6 +908,7 @@ protected WriteInEventLoopCallback newObject(Handle<WriteInEventLoopCallback> ha
                     log.info("[{}] [{}] Closed Producer", topic, producerName);
                     setState(State.Closed);
                     pendingMessages.forEach(msg -> {
+                        client.getMemoryLimitController().releaseMemory(msg.uncompressedSize);
```

Comment:
Do you know if we should also call `msg.sendComplete(ex);` in this block? It seems like these messages would have failed, as the producer is closed and these are pending messages. (This comment is independent of your PR. I'm just asking because I noticed it in the surrounding code while reviewing your PR.)

Reply:
You are right, we should also call `msg.sendComplete(ex)` too,  I can fix it after this PR being `merged`